internal/common/conversion/type_conversion.go

```diff
@@ -56,3 +56,10 @@ func IsStringPresent(strPtr *string) bool {
 func MongoDBRegionToAWSRegion(region string) string {
 	return strings.ReplaceAll(strings.ToLower(region), "_", "-")
 }
+
+func NonEmptySliceToSlicePtr[T any](v []T) *[]T {
```

Comment:
we will likely have a similar approach in CFN and CLI for this version upgrade, could be worth incorporating within SDK as a helper function and mention in upgrade guide.

Reply:
@gssbzn what do you think?

on the other hand I think I'll be able to delete most of the uses of this function in next PRs